test/cql/counters_test.cql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE ks.tbl_cnt (pk int PRIMARY KEY, c counter);
+
+-- insert some values
+UPDATE ks.tbl_cnt SET c = c+1 WHERE pk = 1;
+UPDATE ks.tbl_cnt SET c = c+2 WHERE pk = 2;
+UPDATE ks.tbl_cnt SET c = c+3 WHERE pk = 3;
+UPDATE ks.tbl_cnt SET c = c+4 WHERE pk = 4;
+
+-- test various filtering options on counter column
+SELECT pk, c FROM ks.tbl_cnt WHERE c < 3 ALLOW FILTERING;
+SELECT pk, c FROM ks.tbl_cnt WHERE c <= 3 ALLOW FILTERING;
+SELECT pk, c FROM ks.tbl_cnt WHERE c > 2 AND pk = 4 ALLOW FILTERING;
+SELECT pk, c FROM ks.tbl_cnt WHERE c >= 3 and pk = 3 ALLOW FILTERING;
+SELECT pk, c FROM ks.tbl_cnt WHERE c in (2,3) ALLOW FILTERING;
+SELECT pk, c FROM ks.tbl_cnt WHERE c = 0 ALLOW FILTERING;
```

Comment:
This is an unsuccessful case for counter = <value>. Could you also add a successful case like c=1, please?
I think it would be also interesting to see:
SELECT pk, c FROM ks.tbl_cnt WHERE c in (-1,0,4,5) ALLOW FILTERING;
SELECT pk, c FROM ks.tbl_cnt WHERE c > 4 ALLOW FILTERING;
SELECT pk, c FROM ks.tbl_cnt WHERE c < 1 ALLOW FILTERING;

Reply:
There are two special cases I would love to see in test as well:
1. Uninitialized counter so you probably need to INSERT INTO ks.tbl_cnt(pk) VALUES(5) I think a value of such counter should be 0 so SELECT pk, c FROM ks.tbl_cnt WHERE c = 0 ALLOW FILTERING; should return it and 
SELECT pk, c FROM ks.tbl_cnt WHERE c >= 1 ALLOW FILTERING; should not.
2. Deleted counter. It should not show up at all. This is probably handled by other code than restriction but let's have a test just to be sure.

Reply:
Done!